ast/Trees.h

```diff
@@ -281,8 +281,7 @@ template <class To> To *cast_tree(TreePtr &what) {
     }
 }
 
-// A variant of cast_tree that preserves the const-ness (if const in, then const out)
-template <class To> To const *cast_tree_const(const TreePtr &what) {
```

Comment:
The original intent here was to allow the pointer itself to be reassigned, but ensure that only const methods were called on the data it points to. I thought that `To const *` achieved this?

Reply:
Also, that behavior was only needed for a specific case that I can't remember at this point, so if it works with it returning `const To *` that seems fine to me :)

Reply:
From DM: these two mean the same thing.